lms/lmsweb/views.py

```diff
@@ -207,24 +207,44 @@ def _create_comment(
     })
 
 
-@webapp.route('/notifications', methods=['GET', 'POST'])
+@webapp.route('/notifications')
 @login_required
 def get_notifications():
-    if request.method == 'POST':
-        if request.json:
-            explicit_id = int(request.json.get('notificationId', 0))
-        else:
-            explicit_id = 0
-        changed = notifications.read(user=current_user, id_=explicit_id)
-        if not changed:
-            return fail(403, 'Invalid notification')
-        return jsonify({'success': True})
-
-    # it's a GET
     response = notifications.get(user=current_user)
     return jsonify(response)
 
 
+@webapp.route('/read')
+@webapp.route('/read/<int:notification_id>')
+@webapp.route('/read/<int:notification_id>/<int:related_id>')
+@login_required
+def read_notification(notification_id=None, related_id=None):
```

Comment:
HTTP get route, to change state and abuse get parameters. :(

I can understand if you are really want to do it quick an dirty, but at least add a comment why or fix it to post method that receives in the body the following fields

I saw a year ago article about what the browser reacts to get params vs post. In TL;DR - the browser can do a get param to show you the 'mini' view in chrome without asking you